adapters/repos/db/index.go

```diff
@@ -669,6 +672,12 @@ func (i *Index) IncomingPutObject(ctx context.Context, shardName string,
 		return err
 	}
 
+	shard, release, err := i.getOrInitLocalShardNoShutdown(ctx, shardName)
+	if err != nil {
+		return ErrShardNotFound
```

Comment:
is was not in the original code but it may be worth returning the error returned by `getOrInitLocalShardNoShutdown`, it would require validating that `ErrShardNotFound` is actually returned

Reply:
right, I will change them asap, to better reflect actual issue. Didn't want to block the release for now

Reply:
resolve to merge